string/string.mbt

```diff
@@ -118,36 +118,234 @@ pub fn iter(self : String) -> Iter[Char] {
   )
 }
 
+/// Removes the around whitespaces of this string.
+pub fn trim(self : String) -> String {
+  let len = self.length()
+  let mut start = 0
+  let mut end = len
+  while start < end && self[start] <= ' ' {
+    start += 1
```

Comment:
can you add commment why `self[start] <= '  ' `works, does other language follow the same semantics

Reply:
The Java JDK was implemented like this. I rethought it, maybe provide `trim`, `trim_left` and `trim_right` with a `trim_chars` parameter, and build `trim_space` on top of them, like Go's api.